app/models/operatingsystems/debian.rb

```diff
@@ -38,13 +38,14 @@ def display_family
     "Debian"
   end
 
-  def self.shorten_description(description)
+  def shorten_description(description)
     return "" if description.blank?
     s = description.dup
     s.gsub!('GNU/Linux', '')
     s.gsub!(/\(.+?\)/, '')
     s.squeeze! " "
     s.strip!
+    s += '.' + minor unless s.include?('.')
```

Comment:
Right now it's hard to see what the benefit is. I assume you want to change `Debian GNU/Linux 10 (buster)` into `Debian 10.3` but I'm wondering if that's needed. Debian has dropped the use of minor versions since a few releases. Is there a reason we want to divert from the OS? Does the description column need to be unique?

Could you also add a test case for this?

Reply:
Description was previously "Debian 10", with the change it's gonna be "Debian 10.3".

Uniquness is a good question, I'd say yes we want it to be unique because that's the field we show in lists. You do not want to see this:

* Debian 10
* Debian 10
* Debian 10
* Debian 10

I vaguely remember that this field is preferred because users want to define their own naming convention e.g. RHEL 7.7 instead RedHat 7.7. So generally speaking, i'd rather keep them unique and do this patch than doing a big change on that.

We actually follow the same naming scheme for Red Hat systems and it makes sense to me. Every minor release of RHEL (I assume Debian too) has a installable tree (kickstart). To be able to install from different version trees, you actually need two OS entries otherwise you'd need to change minor version before every provisioning.

I will add a test case for this of couse.

Reply:
@ekohl wait there is a test already, can you elaborate what you want me to do?

https://github.com/theforeman/foreman/pull/7482/files#diff-f6c009374a71e00e01084f53f35f47ba